src/vos/tests/csum_extent_tests.c

```diff
@@ -942,3 +942,78 @@ void csum_invalid_input_tests(void **state)
 	free(read_data_buf);
 	free(csum_buf_1);
 }
+
+/* Inject fault on the update and fetch.
+* The update will initialize checksum value to zero.
+* Fetch operation will return random values in checksum field.
+* Test will compare the update and fetch output for pass/fail
+*/
+void
+csum_fault_injection_multiple_extents_tests(void **state)
```

Comment:
Does this test verify more logic than what's already tested in your changes to io_update_and_fetch_dkey? Honestly, I would rather have more dedicated tests like this than adding additional complexity to other tests, but io_update_and_fetch_dkey does cover single values as well. Should we add specific test cases for single value checksums? Thoughts?

Reply:
The io_update_fetch_dkey single value coverage should be fine for now. In future, we can add separate test case under csum_extent_tests.c file. All changes in vos_io.c should be able still handle injecting faults.